app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java

```diff
@@ -17,11 +22,15 @@
     //hold pixels where canvas will be drawn
     private Bitmap bitmap;
 
+    private Stack<Bitmap> currentStack = new Stack<>();
+    private Stack<Bitmap> undoneStack = new Stack<>();
+
     private float previousX;
     private float previousY;
     private CharcoalTool charcoalTool;
     private EraseTool eraseTool;
     private Tool currentTool;
+    private boolean firstDraw = true;
```

Comment:
Why do you need this variable?

Reply:
It's possible it will be removed once I finish the undo feature. 
But currently, it is used to ensure the user's drawing is actively displayed on the screen (not just on ACTION_UP).

[My onDraw method only displays what's stored in the currentStack of bitmaps. Initially, the stack is empty. So, I check this variable to see if I should print the bitmap so the screen does not remain empty.]